pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -225,11 +225,11 @@ def _evaluation_step(self, batch: Any, batch_idx: int, dataloader_idx: int) -> O
         if self.trainer.testing:
             self.trainer.lightning_module._current_fx_name = "test_step"
             with self.trainer.profiler.profile("test_step"):
-                output = self.trainer.accelerator.test_step(step_kwargs)
+                output = self.trainer.accelerator.test_step(*step_kwargs.values())
```

Comment:
shouldn't this be `**step_kargs`?

Reply:
This works because `values()` is an iterable that returns the pairs, right?

```
def foo(a=1, b=2):
    print(a, b)
    
args = dict(a=3, b=4)
foo(*args.values())
3 4
```


But `**` wouldn't work on that.

Reply:
@awaelchli not on the `values` but on the dict itself. The issue is that the order of `values` is not guaranteed to be consistent with the order of adding them to the dict for python<3.7 and even then we can't be sure we always add them in the correct order. So probably **step_kwargs is a bit safer as it unrolls them to keyword arguments (where the order does not matter). 

@carmocca I agree, however with the `*step_kwargs.values()` it is the way we currently do it.

Reply:
It's not about order because we use OrderedDict

It's actually because we support different names for the arguments, see they NEED to be passed positionally, e.g:

`def test_step(self, b, nb, dl_idx)`